api/server/router/image/image_routes.go

```diff
@@ -126,6 +126,7 @@ func (s *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrite
 				if err := json.NewDecoder(authJSON).Decode(authConfig); err != nil {
 					// for a pull it is not an error if no auth was given
 					// to increase compatibility with the existing api it is defaulting to be empty
+					errors.Errorf("Failed decoding the X-Registry-Auth in request header")
```

Comment:
You just create an error and you're doing nothing with it.
Also, Why are you using `Errorf` while you have `New`?

Reply:
we will catch this error on CLI , so I guess it is necessary !

Reply:
The comment is no longer correct as well, so I guess it would need to be something like;

```patch
diff --git a/api/server/router/image/image_routes.go b/api/server/router/image/image_routes.go
index dabab3bcf..feea9669e 100644
--- a/api/server/router/image/image_routes.go
+++ b/api/server/router/image/image_routes.go
@@ -25,6 +25,20 @@ import (
        "golang.org/x/net/context"
 )

+type validationError struct {
+       cause error
+}
+
+func (e validationError) Error() string {
+       return e.cause.Error()
+}
+
+func (e validationError) Cause() error {
+       return e.cause
+}
+
+func (validationError) InvalidParameter() {}
+
 func (s *imageRouter) postCommit(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
        if err := httputils.ParseForm(r); err != nil {
                return err
@@ -124,9 +138,7 @@ func (s *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrite
                        if authEncoded != "" {
                                authJSON := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authEncoded))
                                if err := json.NewDecoder(authJSON).Decode(authConfig); err != nil {
-                                       // for a pull it is not an error if no auth was given
-                                       // to increase compatibility with the existing api it is defaulting to be empty
-                                       authConfig = &types.AuthConfig{}
+                                       return errors.Wrap(validationError{err}, "malformed X-Registry-Auth")
                                }
                        }
                        err = s.backend.PullImage(ctx, image, tag, platform.OS, metaHeaders, authConfig, output)
@@ -148,20 +160,6 @@ func (s *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrite
        return nil
 }

-type validationError struct {
-       cause error
-}
-
-func (e validationError) Error() string {
-       return e.cause.Error()
-}
-
-func (e validationError) Cause() error {
-       return e.cause
-}
-
-func (validationError) InvalidParameter() {}
-
 func (s *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
        metaHeaders := map[string][]string{}
        for k, v := range r.Header {
@@ -179,8 +177,7 @@ func (s *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter,
                // the new format is to handle the authConfig as a header
                authJSON := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authEncoded))
                if err := json.NewDecoder(authJSON).Decode(authConfig); err != nil {
-                       // to increase compatibility to existing api it is defaulting to be empty
-                       authConfig = &types.AuthConfig{}
+                       return errors.Wrap(validationError{err}, "malformed X-Registry-Auth")
                }
        } else {
                // the old format is supported for compatibility if there was no authConfig header
@@ -344,9 +341,7 @@ func (s *imageRouter) getImagesSearch(ctx context.Context, w http.ResponseWriter
        if authEncoded != "" {
                authJSON := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authEncoded))
                if err := json.NewDecoder(authJSON).Decode(&config); err != nil {
-                       // for a search it is not an error if no auth was given
-                       // to increase compatibility with the existing api it is defaulting to be empty
-                       config = &types.AuthConfig{}
+                       return errors.Wrap(validationError{err}, "malformed X-Registry-Auth")
                }
        }
        for k, v := range r.Header {
```

(I moved the definition of the `validationError` to the top of the file; not necessary. but is a bit cleaner to define it at the top now)